lib/wellies_web/controllers/weather_controller.ex

```diff
@@ -1,8 +1,44 @@
 defmodule WelliesWeb.WeatherController do
   use WelliesWeb, :controller
 
-  def weather(conn, %{"location" => location} = _params) do
-    render(conn, "weather.html", location: location)
+  def weather(conn, %{"location_id" => location_id} = _params) do
+    temperature = get_current_temp(location_id)
+    render(conn, "weather.html", temperature: temperature)
+  end
+
+  def get_current_temp(id) do
+    response = request_12h_weather_in(id)
+    body = get_body(response)
+    get_temperature(body)
+  end
+
+  def request_12h_weather_in(id) do
+    Application.ensure_all_started(:inets)
```

Comment:
Could this be extracted out into a file? Like all requests made to the openweathermap? So it's not in the controller?

Reply:
thanks for the comment! I applied your feedback! :)